CONTRIBUTING.md

```diff
@@ -0,0 +1,92 @@
+# How to contribute
+
+Operator Controller is an Apache 2.0 licensed project and accepts contributions via GitHub pull requests (PRs).
+
+This document outlines some conventions on commit message formatting, contact points for developers, and other resources
+to help Operator Controller contributors.
+
+## Communication
+
+- Email: [operator-framework-olm-dev](mailto:operator-framework-olm-dev@googlegroups.com)
+- Slack: [#olm-dev](https://kubernetes.slack.com/archives/C0181L6JYQ2)
+- Google Group: [olm-gg](https://groups.google.com/g/operator-framework-olm-dev)
+- Weekly in Person Working Group Meeting: [olm-wg](https://github.com/operator-framework/community#operator-lifecycle-manager-working-group) 
+
+## Getting started
+
+- Fork the repository on GitHub.
+- Clone onto your local development machine via `git clone https://github.com/{$GH-USERNAME}/operator-controller`.
+- Add the operator-framework/operator-controller as an upstream.
+  remote: `git remote add upstream https://github.com/operator-framework/operator-controller`.
+- Create a new branch from the default `main` branch and begin development.
+
+## Contribution flow
+
+Any new contribution should be accompanied by a new or existing issue. This issue can help track work, discuss the
+design and implementation, and help avoid wasted efforts of multiple people working on the same issue. Trivial changes,
+like fixing a typo in the documentation, do not require the creation of a new issue.
+
+### Small Contributions
+
+For simple contributions, this is a rough outline of what a contributor's workflow looks like:
+
+- Identify or create a GitHub Issue.
+- Create a topic branch from where to base the contribution. This is usually the main branch.
+- Make commits of logical units.
+- Make sure commit messages are in the proper format (see below).
+- Push changes in a topic branch to a personal fork of the repository.
```

Comment:
We possibly need to mention test/coverage expectations with codecov.

Reply:
I don't think that we have any hardcore code coverage expectations, but I'll call out the need for tests.